### PR TITLE
Make group maintainers visible in the WebUI

### PR DESCRIPTION
**Pull request by bgeuken**
They were previously only exposed via the API.

This was needed to be able show and edit the maintainer rights of users.
The group user edit and show page are now merged, which also solves the
different appearance of group users on both pages.

The "Actions" column in the group index view became superfluous because
it only contained a link to the, now removed, user edit view.
A link to the show page is still there.

Users can now see who is a maintainer of a group. Group maintainers can
change maintainer rights of other users.

Before:

Now:

And for smaller screen devices:

### CODE REVIEW

File: src/api/app/views/webui2/webui/groups/_group_members.html.haml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+- write_access = policy(group).update?
+
+- if write_access
+  = link_to('#', class: 'nav-link', data: { toggle: 'modal', target: '#add-group-user-modal' }) do
+    %i.fas.fa-plus-circle.text-success
+    Add User
```
Comment (bgeuken) [inline src/api/app/views/webui2/webui/groups/_group_members.html.haml]: This currently expands over the full screen. I haven't had time to debug this yet. If someone knows a fix for this, please let me know.

**Comment by bgeuken** [comment]
[Introduce group self-service](https://trello.com/c/xGs0ho4z/1080-introduce-group-self-service)

**Comment by bgeuken** [comment]
> I think that the close arrow so close to the name looks a bit weird, apart from that I like the design more than the previous one :+1:

Me too, but I didn't figure a way to do this:laughing: Not sure if I find the time to research this till end of sprint.

File: src/api/app/views/webui2/webui/groups/_add_group_user_modal.html.haml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+.modal.fade#add-group-user-modal{ tabindex: -1, role: 'dialog', aria: { labelledby: 'add-group-user-modal-label', hidden: true } }
+  .modal-dialog.modal-dialog-centered{ role: 'document' }
+    .modal-content
+      = form_tag(group_user_index_path(group_title: group.title)) do
+        = hidden_field_tag 'group_title', group.title
+        .modal-header
+          %h5.modal-title#add-group-user-modal-label Add User
+        .modal-body
+          = render partial: 'webui/autocomplete', locals: { html_id: 'user_login', label: 'Replace the group with:',
```
[the remark this answers is not in the document]
Comment (bgeuken) [reply src/api/app/views/webui2/webui/groups/_add_group_user_modal.html.haml]: Oups, fixed it.

**Comment by Ana06** [comment]
@bgeuken 

> Me too, but I didn't figure a way to do thislaughing Not sure if I find the time to research this till end of sprint.

maybe moving it to the right makes it.

**Comment by bgeuken** [comment]
@saraycp @Ana06 Updated. Please have another look. I didn't change the position of the delete icon, since I didn't find the time for that. If you know how to fix this feel free to push to this PR.

**Comment by dmarcoux** [comment]
Issues with this:

- It's not possible anymore to edit a group (changing its title for example if you happen to have a typo)
- Adding members is not possible anymore when creating a group
- We aren't consistent in the naming for controllers, routes and in the views. We sometimes use members and in other cases, we use users.

**Comment by saraycp** [comment]
IMHO @dmarcoux is right, but I think we can re-think this solution after the deliverable.

**Comment by dmarcoux** [comment]
@Ana06 Please have another look. The delete icon is now to the right (with the CSS class `float-right`)

**Comment by Ana06** [comment]
I would still move the arrow to the right though

**Comment by saraycp** [comment]
@dmarcoux I propose to change this

Screenshot-2019-4-1 Open Build Service(7)

by this

Screenshot-2019-4-1 Open Build Service(6)

Patch:

[delete_icon.zip](https://github.com/openSUSE/open-build-service/files/3030353/delete_icon.zip)

**Comment by dmarcoux** [comment]
@saraycp I like your proposal :+1:.

Regarding the specs, they pass locally.... so I guess it's probably caused by Capybara not waiting.

**Comment by bgeuken** [comment]
To be fair that was basically what I suggested at the beginning:-), move one step back to the data table view and integrate the maintainer toggle there. And then spend some time (after the sprint) with hashing out some design that covers all aspects.

**Pull request merged**